addons/project/models/project.py

```diff
@@ -0,0 +1,832 @@
+
+        result = super(Task, self).write(vals)
+
+        if any(item in vals for item in ['stage_id', 'remaining_hours', 'user_id', 'kanban_state']):
+            self._store_history()
+        return result
+
+    @api.multi
+    def copy_data(self, default=None, context=None):
+        self.ensure_one()
+        if default is None:
+            default = {}
+        if not default.get('name'):
+            default['name'] = _("%s (copy)") % self.name
+        if 'remaining_hours' not in default:
+            default['remaining_hours'] = self.planned_hours
+
+        return super(Task, self).copy_data(default)[0]
+
+    # Override view according to the company definition
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        # read uom as admin to avoid access rights issues, e.g. for portal/share users,
+        # this should be safe (no context passed to avoid side-effects)
+        product_uom = self.env.user.company_id.project_time_mode_id
+        project_data = super(Task, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
+
+        # read uom as admin to avoid access rights issues, e.g. for portal/share users,
+        # this should be safe (no context passed to avoid side-effects)
+        uom_hour = self.env.ref('product.product_uom_hour', raise_if_not_found=False)
+        if not product_uom or not uom_hour or product_uom.id == uom_hour.id:
+            return project_data
+
+        eview = etree.fromstring(project_data['arch'])
+
+        # if the project_time_mode_id is not in hours (so in days), display it as a float field
+        def _check_rec(eview):
+            if eview.attrib.get('widget','') == 'float_time':
+                eview.set('widget','float')
+            for child in eview:
+                _check_rec(child)
+            return True
+
+        _check_rec(eview)
+
+        project_data['arch'] = etree.tostring(eview)
+
+        # replace reference of 'Hours' to 'Day(s)'
+        for project_field in project_data['fields']:
+            # TODO this NOT work in different language than english
+            # the field 'Initially Planned Hours' should be replaced by 'Initially Planned Days'
+            # but string 'Initially Planned Days' is not available in translation
+            if 'Hours' in project_data['fields'][project_field]['string']:
+                project_data['fields'][project_field]['string'] = project_data['fields'][project_field]['string'].replace('Hours', product_uom.name)
+        return project_data
+
+    @api.model
+    def get_empty_list_help(self, help):
+        context = dict(self.env.context)
+        context['empty_list_help_id'] = context.get('default_project_id')
+        context['empty_list_help_model'] = 'project.project'
+        context['empty_list_help_document_name'] = _("tasks")
+        return super(Task, self.with_context(context)).get_empty_list_help(help)
```

Comment:
use `self.with_context(**kwargs)` instead of manually copying the context to update a few keys.